engine/tests/ElggUserTest.php

```diff
@@ -177,6 +179,9 @@ public function testElggUserRemoveAdmin() {
 
 		$this->assertTrue($this->user->removeAdmin());
 
+		// need to save user to propagate attributes to the database
+		$this->user->save();
```

Comment:
There was a bug in `ElggEntity::delete()`. Cache was invalidated, but it was being repopulated in one of the consequent delete operations (not sure which one - metadata, relationships or other). I have moved cache invalidation closer to the DB delete query.